From: Werner Koch <wk@gnupg.org>
Date: Wed, 30 Jan 2019 14:40:26 +0100
Subject: gpg: Emit an ERROR status if no key was found with --list-keys.

* g10/keylist.c (list_one): Emit status line.
--

Signed-off-by: Werner Koch <wk@gnupg.org>
(cherry picked from commit 140fda8c61422ec055c3f7e214cc35706c4320dd)
(cherry picked from commit 14ea581a1c040b53b0ad4c51136a7948363b1e4b)
---
 g10/keylist.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/g10/keylist.c b/g10/keylist.c
index 66b03bb..262ea8d 100644
--- a/g10/keylist.c
+++ b/g10/keylist.c
@@ -610,6 +610,7 @@ list_one (ctrl_t ctrl, strlist_t names, int secret, int mark_secret)
     {
       log_error ("error reading key: %s\n", gpg_strerror (rc));
       getkey_end (ctrl, ctx);
+      write_status_error ("keylist.getkey", rc);
       return;
     }
 
